test/tests/functional/pbs_cray_vnode_pool.py

```diff
@@ -0,0 +1,325 @@
+
+    """
+    This test suite tests how PBS makes use of node attribute "vnode_pool"
+    It expects at least 2 moms to be specified to it while executing.
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+
+        # The moms provided to the test may have unwanted vnodedef files.
+        if self.moms.values()[0].has_vnode_defs():
+            self.moms.values()[0].delete_vnode_defs()
+        if self.moms.values()[1].has_vnode_defs():
+            self.moms.values()[1].delete_vnode_defs()
+
+        # Check if vnodes exist before deleting nodes.
+        # Clean all default nodes because each test case will set up nodes.
+        try:
+            self.server.status(NODE)
+            rc = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+            self.assertEqual(rc, 0)
+        except PbsStatusError as e:
+            self.assertEquals(
+                e.msg[0], "/opt/pbs/bin/pbsnodes: Server has no node list")
+
+    def test_invalid_values(self):
+        """
+        Invalid vnode_pool values shall result in errors.
+        """
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+
+        attr_A = {'vnode_pool': '-1'}
+        try:
+            self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA,
+                                attrib=attr_A, expect=False)
+        except PbsManagerError as e:
+            self.assertEquals(
+                e.msg[1], "qmgr: Error (15014) returned from server")
+
+        attr_A = {'vnode_pool': '0'}
+        try:
+            self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA,
+                                attrib=attr_A, expect=False)
+        except PbsManagerError as e:
+            self.assertEquals(
+                e.msg[1], "qmgr: Error (15014) returned from server")
+
+        attr_A = {'vnode_pool': 'a'}
+        try:
+            self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA,
+                                attrib=attr_A, expect=False)
+        except PbsManagerError as e:
+            self.assertEquals(
+                e.msg[1], "qmgr: Error (15014) returned from server")
+
+    def test_two_moms_single_vnode_pool(self):
+        """
+        Same vnode_pool for two moms shall result in one mom being the
+        inventory mom and the other the non-inventory mom.
+        The inventory mom goes down (e.g. killed).
+        Compute nodes remain up even when the inventory mom is killed,
+        since another mom is reporting them.
+        Check that a new inventory mom is listed in the log.
+        Bring up killed mom.
+        """
+        self.server.manager(MGR_CMD_SET, SERVER, {"log_events": -1})
+
+        found_in_momA = 0
+        found_in_momB = 0
+        self.momA = self.moms.values()[0]
+        try:
+            self.momB = self.moms.values()[1]
+        except IndexError:
+            self.assertTrue(
```

Comment:
The test requires two moms and if it were to silently skip the test then we would have no failure or error to indicate that the test used less than two moms.